// *****************************************************************************
//
// This file is part of a MASA library or program.
// Refer to the included end-user license agreement for restrictions.
//
// Copyright (c) 2010 MASA Group
//
// *****************************************************************************

package sword;

/** @struct ServicesDescription
    @brief A message sent by the Dispatcher to announce available services.
    
    This message is sent just after a client connection request. It provides
    a list of strings identifying the services available on the Dispatcher.
    Services are optional modules such as indicator computation (for 
    After/During Action Review).
*/
message ServicesDescription {
    /** @property services
        @brief The list of available services identifiers.
    */
    repeated string services = 1;
}

/** @struct DispatcherToClient
    @brief A facade for all available messages going from Dispatcher to client.
    
    This is the actual message sent over the network.
*/
message DispatcherToClient {
    /** @struct Content
        @brief The possible messages contained in a Dispatcher to client message.
        @warning Each Content message must contain exactly one of the possible messages.
    */
    message Content {
        optional ServicesDescription services_description = 1; //!< A services description message
    }
    /** @property message
        @brief A field containing one of the message to be sent.
    */
    required Content message = 1;
}
